### DISCUSSION TITLE
Flickering when commiting in lazygit and tmux

### DISCUSSION THREAD

**Discussion by juanibiapina** [ghostty-org/ghostty#3801] [Q&A]
When writing a commit message from within `tmux` and `lazygit` (pressing the `c` shortcut), I get a lot of flickering. Specifically, on every key press I can see a cursor blinking somewhere else on the screen. This doesn't happen outside of tmux.

Has anyone seen this too?

Tested cases:
- `alacritty -> lazygit`: working :heavy_check_mark:
- `ghostty -> lazygit`: working :heavy_check_mark: 
- `alacritty -> tmux -> lazygit`: flickering :warning: 
- `ghostty -> tmux -> lazygit`: flickering :warning: 

Possibly related:
- similar report for lazygit: https://github.com/jesseduffield/lazygit/issues/3434
- synchronized updates issue on tmux: https://github.com/tmux/tmux/issues/3325

Updates:
- 03.01.2025: I managed to reproduce in alacritty.

**Comment by 00-kat** [comment]
If I remember correctly, this happens because Ghostty is *faster* at rendering, so it's rendering more frequently, which happens to coincide more often with when the TUI is drawing itself.  The best way to solve this would be for the respective TUIs to support the [Synchronised Output] extension.

[Synchronised Output]: https://gist.github.com/christianparpart/d8a62cc1ab659194337d73e399004036

**Reply by 00-kat** [reply]
I just found a message by @jcollie on the Ghostty Discord about this as well ([message]).
> typically flickering like this is because Ghostty is _faster_ than other terminals in drawing/updating the screen. The only real solution is to get the CLI app to support mode 2026 synchronized output (<https://gist.github.com/christianparpart/d8a62cc1ab659194337d73e399004036>). That allows the CLI to tell Ghostty (or any other terminal that supports it - many do) when to update the screen to eliminate tearing/flicker.

[message]: https://discord.com/channels/1005603569187160125/1314321983915425843/1314338022531010692

**Comment by juanibiapina** [comment]
Possibly related:
- same report on lazygit: https://github.com/jesseduffield/lazygit/issues/3434
- synchronized updates issue on tmux: https://github.com/tmux/tmux/issues/3325
